onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -206,9 +203,13 @@ Status Conv<T>::UpdateState(OpKernelContext* context, bool bias_expected) const
       dilations.push_back(1);
     }
 
-    if (w_dims_changed) {
+    if (w_dims_changed)
       ORT_RETURN_IF_ERROR(s_.w_desc.Set(w_dims, CudnnTensor::GetDataType<CudaT>()));
+
+    if (s_.Y->Shape().Size() == 0) {
```

Comment:
Maybe add a comment here stating that bailing out eary has to happen after the weight dims have been cached for the next Run() ?

Reply:
Will add it with the test as you suggested